dbms/src/Server/Bootstrap.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/Exception.h>
+#include <Common/RedactHelpers.h>
+#include <Storages/Page/V3/Universal/UniversalPageIdFormatImpl.h>
+#include <Storages/Page/V3/Universal/UniversalPageStorage.h>
+#include <raft_serverpb.pb.h>
+
+namespace DB
+{
+
+std::optional<raft_serverpb::StoreIdent>
+tryGetStoreIdent(const UniversalPageStoragePtr & wn_ps)
```

Comment:
I'd better make getting meta states such like StoreIdent/RaftApplyState/RegionLocalState common. This is because in v2, they are stored in RaftEngine rather than KVEngine. So in unips we must access them by a different prefix.
I think maybe a better way is we can call an FFI interface, and let Proxy to fetch the result to decouple the complex logic.

Reply:
`RaftApplyState`/`RegionLocalState` is for a specified region, I think they should not belong here. This file and function is mainly for checking whether a cluster is already bootstrap.
And later we may split `TMTContext` into smaller parts and do the bootstrap more earlier than now.

If we want to adapt `RaftApplyState`/`RegionLocalState` with raft-store V2, we could do it in later PRs.